File: Modules/Common/Equipments/SIE.Web.Fixtures/Accounts/Commands/EditSaveCodeAccCommand.js
SIE.defineCommand('SIE.Web.Fixtures.Accounts.Commands.EditSaveCodeAccCommand', {
    extend: 'SIE.cmd.FormSave',
    meta: { text: "保存", group: "edit", iconCls: "icon-SaveEntity icon-blue" },
    /**
    * 保存
    * @param {} view
    */
    doSave: function (view) {
        var me = this;
        me.view = view;
        var data = view.getCurrent().data;
        var indata = {};
        indata.Data = Ext.encode(data);
        view.execute({
            data: indata,
            success: function (res) {
                var errMsg = res.Result;
                if (errMsg !== '') {
                    SIE.Msg.showError(errMsg);
                    return;
                } else {
                    SIE.Msg.showInstantMessage('保存成功！', '提示', 3);
                    entity.markSaved();
                    me.view.syncCmdState();
                    window.setTimeout(function () {
                        CRT.Event.fire("SIE.Fixtures.Fixtures.Accounts.FixtureCodeAccount_refresh");
                        CRT.Workbench.closeCurrentTab();
                    }, 3000);
                }
            }
        });
    }
});

File: Modules/Common/Equipments/SIE.Web.Fixtures/Accounts/Commands/EditSaveCodeAccCommand.cs
using SIE.Fixtures;
using SIE.Fixtures.Fixtures.Accounts;
using SIE.Web.Command;
using System;

namespace SIE.Web.Fixtures.Accounts.Commands
{
    /// <summary>
    /// 保存编码类工治具台帐
    /// </summary>
    [JsCommand("SIE.Web.Fixtures.Accounts.Commands.EditSaveCodeAccCommand")]
    public class EditSaveCodeAccCommand : ViewCommand<ViewArgs>
    {
        /// <summary>
        /// 执行保存操作
        /// </summary>
        /// <param name="args">args</param>
        /// <param name="scope">scope</param>
        /// <returns>执行结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            string errMsg = string.Empty;
            var account = args.Data.ToJsonObject<FixtureCodeAccount>();
            try
            {
                errMsg = RT.Service.Resolve<CoreFixtureController>().SaveFixtureCodeAccount(account);
            }
            catch (Exception ex)
            {
                errMsg = ex.Message;
            }

            return errMsg;
        }
    }
}
